am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/importer/DefaultPermissionIntegrationTest.java

```diff
@@ -91,35 +106,351 @@ void shouldOverwriteExistingRecordWhenEntryIsAddedASecondTime() {
             "", roleName, READ.getValue());
 
-        final Instant localDateTime = roleBasedAccessRecord.getAccessManagementAudit().getLastUpdate();
+        final Instant localDateTime = roleBasedAccessRecord.getLastUpdate();
         assertThat(roleBasedAccessRecord).isNotNull();
-        assertThat(roleBasedAccessRecord.getAccessManagementAudit().getCallingServiceName()).isNotNull();
-        assertThat(roleBasedAccessRecord.getAccessManagementAudit().getCallingServiceName())
+        assertThat(roleBasedAccessRecord.getCallingServiceName()).isNotNull();
+        assertThat(roleBasedAccessRecord.getCallingServiceName())
             .isEqualTo(CALLING_SERVICE_NAME_FOR_INSERTION);
         assertThat(localDateTime).isNotNull();
 
         //Update Audit
-        audit = AccessManagementAudit.builder().lastUpdate(Instant.now())
-            .callingServiceName(CALLING_SERVICE_NAME_FOR_UPDATES).build();
         service.grantDefaultPermission(
-            grantDefaultPermissionForResourceWithAudit(roleName, resourceDefinition, ImmutableSet.of(READ), audit));
+            grantDefaultPermissionForResourceWithAudit(roleName, resourceDefinition, ImmutableSet.of(READ),
+                CALLING_SERVICE_NAME_FOR_UPDATES, null));
+
         roleBasedAccessRecord = databaseHelper.getDefaultPermissionsForAudit(resourceDefinition, "",
             roleName, READ.getValue());
         assertThat(roleBasedAccessRecord).isNotNull();
-        assertThat(roleBasedAccessRecord.getAccessManagementAudit().getCallingServiceName()).isNotNull();
-        assertThat(roleBasedAccessRecord.getAccessManagementAudit().getCallingServiceName())
+        assertThat(roleBasedAccessRecord.getCallingServiceName()).isNotNull();
+        assertThat(roleBasedAccessRecord.getCallingServiceName())
             .isEqualTo(CALLING_SERVICE_NAME_FOR_UPDATES);
-        assertThat(roleBasedAccessRecord.getAccessManagementAudit().getLastUpdate()).isNotNull();
-        assertThat(roleBasedAccessRecord.getAccessManagementAudit().getLastUpdate()).isNotEqualTo(localDateTime);
+        assertThat(roleBasedAccessRecord.getLastUpdate()).isNotNull();
+        assertThat(roleBasedAccessRecord.getLastUpdate()).isNotEqualTo(localDateTime);
+    }
+
+    @Test
+    void whenAddedDefaultPermissionsShouldAuditDefaultPermissionsAndAttributesRecords() {
+        service.addRole(roleName, RESOURCE, PUBLIC, ROLE_BASED);
+        service.grantDefaultPermission(
+            grantDefaultPermissionForResourceWithAudit(roleName, resourceDefinition, ImmutableSet.of(READ),
+                CALLING_SERVICE_NAME_FOR_INSERTION, CHANGED_BY_NAME_FOR_INSERTION));
+
+        List<RoleBasedAccessAuditRecord> accessAuditRecords = databaseHelper.getDefaultPermissionsAuditRecords(
+            resourceDefinition, "", roleName, READ.getValue());
+        assertThat(accessAuditRecords).isNotNull();
+
+        //Audit flag on
+        if (TRUE.toString().equalsIgnoreCase(PropertyReader.getPropertyValue(AUDIT_REQUIRED))) {
```

Comment:
having a conditional inside the test means that only part of the test will be executed. Is there a way to split this into two different tests that cover each part of the conditional? (ie. AUDIT_REQUIRED=true, AUDIT_REQUIRED=false)